client/src/main/java/com/spectralogic/tpfr/client/ClientImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.spectralogic.tpfr.client;
+
+import com.spectralogic.tpfr.api.ServerService;
+import com.spectralogic.tpfr.api.ServerServiceFactory;
+import com.spectralogic.tpfr.api.ServerServiceFactoryImpl;
+import com.spectralogic.tpfr.api.response.IndexStatus;
+import com.spectralogic.tpfr.api.response.errors.GeneralError;
+import com.spectralogic.tpfr.api.response.errors.GeneralErrorResponseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+public class ClientImpl implements Client {
+    private static final Logger LOG = LoggerFactory.getLogger(ClientImpl.class);
+
+    private final ServerServiceFactory serverServiceFactory;
+
+    public ClientImpl(final String endpoint){
+        this.serverServiceFactory = new ServerServiceFactoryImpl();
```

Comment:
It's better to pass in all the dependent services into the client rather than being created by it.  This will make the clientImpl testable by allowing us to inject our own mocked ServerServiceFactory instance